### ISSUE TITLE
devtools automationProtocol: execute/executeAsync removing 7 characters from scripts, breaking them.

### ISSUE THREAD

**Issue by dylang** [webdriverio/webdriverio#4658]
**Environment (please complete the following information):**
 - **WebdriverIO version:** 5.15.1
 - **Mode:** standalone
 - **If WDIO Testrunner, running sync/async:** [e.g. sync/async]
 - **Node.js version:** 10
 - **NPM version:** yarn 1.19.1
 - **Browser name and version:** Chrome 76
 - **Platform name and version:** MacOS
 - **Additional wdio packages used (if applicable):**  devtools

**Config of WebdriverIO**

```
automationProtocol: 'devtools',
```

**Describe the bug**

```ts
const el = await this.webdriverio.$(selector);
await el.waitForExist();

// Move the "mouse" to the object.
// This is necessary for items that require hover/mouseenter before a click is recognized.
await el.scrollIntoView();
await el.moveTo(); // fails here
```

## Error

```
Unexpected token ':'
```

## Code being executed:
```
return { scrollX: this.pageXOffset, scrollY: this.pageYOffset }
```
This is from `webdriverio`:
https://github.com/webdriverio/webdriverio/blob/master/packages/webdriverio/src/utils.js#L361

It fails because `executeScript` is manipulating the string without checking to see if the change is valid:
```ts
const result = eval(script.slice(7))
```
https://github.com/webdriverio/webdriverio/blob/master/packages/devtools/src/scripts/executeScript.js#L18,

This results in attempting to run:
```ts
eval('{ scrollX: this.pageXOffset, scrollY: this.pageYOffset }');
```

This is not valid JS.

It *should* be
```ts
eval('({ scrollX: this.pageXOffset, scrollY: this.pageYOffset })');
```

(notice the parens)

**To Reproduce**

*Example 1.* see above.

*Example 2.* `await browser.execute('console.log("hello")');`

```
    Evaluation failed: SyntaxError: Unexpected token '.'
        at __puppeteer_evaluation_script__:14:30
```

*Example 3.* `await browser.execute('          return "string";');`

```
    Evaluation failed: SyntaxError: Illegal return statement
        at __puppeteer_evaluation_script__:14:30
```

*Example 4.* `await browser.execute('/* test */ console.log("test")');`

```
    Evaluation failed: SyntaxError: Unexpected token '*'
        at __puppeteer_evaluation_script__:14:30
```

*Example 5.* `executeAsync` has this issue as well.

**Expected behavior**

* `execute` and `executeAsync` should never blindly remove characters. Either require developers to provide strings already formatted correctly (and fix code [in webdriverio](https://github.com/webdriverio/webdriverio/blob/master/packages/webdriverio/src/utils.js#L361)), or carefully examine strings before making any changes.
* Ideally, any manipulation of the script should be done before sending to the chrome debug API so that the exception and logs show the actual script being executed. This was difficult to trace because the code looks fine.

**Log**

`puppeteer:protocol SEND ►`
```json
{
  "sessionId": "90ED379684DB047F703177EF7F0F78E4",
  "method": "Runtime.callFunctionOn",
  "params": {
    "functionDeclaration": "(_, script, dataProperty, dataFlag, ...args) => {\n  window.arguments = args;\n    console.log('vvvvvvvvvvvvvvvvvvvvvvvvvvvv')\n  console.log('script.slice(7)', script.slice(7));\n  try {\n      const result = eval(script.slice(7));\n\n  } catch (err) {\n\n      console.log('ERR', err);\n      console.log(script);\n      console.log('==========================')\n  }\n  const result = eval(script.slice(7));\n  console.log('result', result);\n    console.log('^^^^^^^^^^')\n  let tmpResult = result instanceof NodeList ? Array.from(result) : result;\n  const isResultArray = Array.isArray(tmpResult);\n  tmpResult = isResultArray ? tmpResult : [tmpResult];\n\n  if (tmpResult.find(r => r instanceof HTMLElement)) {\n    tmpResult = tmpResult.map((r, i) => {\n      if (r instanceof HTMLElement) {\n        const dataPropertyValue = `${dataFlag}_${i}`;\n        r.setAttribute(dataProperty, dataPropertyValue);\n        return dataPropertyValue;\n      }\n\n      return result;\n    });\n  }\n\n  return isResultArray ? tmpResult : tmpResult[0];\n}\n//# sourceURL=__puppeteer_evaluation_script__\n",
    "executionContextId": 7,
    "arguments": [
      {
        "objectId": "{\"injectedScriptId\":7,\"id\":22}"
      },
      {
        "value": "return { scrollX: this.pageXOffset, scrollY: this.pageYOffset };"
      },
      {
        "value": "data-devtoolsdriver-fetchedElement"
      },
      {
        "value": "__executeElement"
      }
    ],
    "returnByValue": true,
    "awaitPromise": true,
    "userGesture": true
  },
  "id": 107
}
```
`puppeteer:protocol ◀ RECV`
```json
{
  "method": "Runtime.consoleAPICalled",
  "params": {
    "type": "log",
    "args": [
      {
        "type": "string",
        "value": "ERR"
      },
      {
        "type": "object",
        "subtype": "error",
        "className": "SyntaxError",
        "description": "SyntaxError: Unexpected token ':'\n    at __puppeteer_evaluation_script__:6:34",
        "objectId": "{\"injectedScriptId\":7,\"id\":23}",
        "preview": {
          "type": "object",
          "subtype": "error",
          "description": "SyntaxError: Unexpected token ':'\n    at __puppeteer_evaluation_script__:6:34",
          "overflow": false,
          "properties": [
            {
              "name": "stack",
              "type": "string",
              "value": "SyntaxError: Unexpected token ':'\n    at __puppeteer_evaluation_script__:6:34"
            },
            {
              "name": "message",
              "type": "string",
              "value": "Unexpected token ':'"
            }
          ]
        }
      }
    ],
    "executionContextId": 7,
    "timestamp": 1571419795871.338,
    "stackTrace": {
      "callFrames": [
        {
          "functionName": "",
          "scriptId": "22",
          "url": "__puppeteer_evaluation_script__",
          "lineNumber": 9,
          "columnNumber": 14
        }
      ]
    }
  },
  "sessionId": "90ED379684DB047F703177EF7F0F78E4"
} 
```

**Additional context**

We need to put all javascript that should execute in browser in a string so that TypeScript doesn't expect it to be typed and our code coverage tooling doesn't attempt to decorate it or count it towards coverage.

**Comment by mgrybyk** [comment]
@dylang mind proposing a PR with fix?

**Comment by dylang** [comment]
Before doing that, I'd like to understand the motivation around `slice(7)` and how it's not already being flagged in tests for breaking methods like `moveTo`.

**Comment by mgrybyk** [comment]
@dylang it happens because we expect regular function not arrow function

**Comment by dylang** [comment]
@mgrybyk Thanks for the suggestion, but that doesn't seem to be it. 

* It will remove `functio` and leave the `n`, so it will still be broken.
* This is a "regular function" and failing: https://github.com/webdriverio/webdriverio/blob/master/packages/webdriverio/src/utils.js#L361

I think it's assuming there's a `return` in the script, but then the code should check for that string before removing the characters. But it still doesn't fix the code in webdriverio that calls it like the example in this message.

**Comment by mgrybyk** [comment]
Yeah, guess you are right, I remember there was some keyword.

**Comment by mgrybyk** [comment]
Anyway moveTo won't work because performActions are currently not supported with devtools protocol

**Comment by dylang** [comment]
> Anyway moveTo won't work because performActions are currently not supported with devtools protocol. 

What about `await page.hover(selector);`?

**Comment by christian-bromann** [comment]
A fix for this was proposed in #4829, let's continue conversation there.

**Comment by dylang** [comment]
Sorry I missed this while on Thanksgiving break.

Thank you for the fix, however it seems to be incomplete. The test cases only cover when there is a `return` statement at the beginning of the line, but not when there are other statements before the `return`:

```ts
const result = await webdriverio.execute<string>(`
  const el = document.querySelector(\`${selector}\`);
   return (el.innerText || el.textContent).trim().replace(/\\s+/g, ' ')
`);
```

Result:

```
   Evaluation failed: SyntaxError: Illegal return statement
        at __puppeteer_evaluation_script__:3:23
```
